cloud/setup/kubernetes.tf

```diff
@@ -0,0 +1,119 @@
+  }
+  spec {
+    parallelism = var.producer_instances
+    template {
+      metadata {}
+      spec {
+        container {
+          name  = "producer-benchmark"
+          image = var.bench_producer_image
+          image_pull_policy = var.image_pull_policy
+          env {
+            name = "KAFKA_BOOTSTRAP_SERVERS"
+            value = confluent_kafka_cluster.perf-test-dedicated.bootstrap_endpoint
+            #value = "localhost:9092"
+          }
+          env {
+            name = "KAFKA_SECURITY_PROTOCOL"
+            value = "SASL_SSL"
+          }
+          env {
+            name = "KAFKA_SASL_MECHANISMS"
+            value = "PLAIN"
+          }
+
+          env {
+            name = "KAFKA_SASL_MECHANISM"
+            value = "PLAIN"
+          }
+          env { # GroupBy(key)
+            name = "AGG_PER_TOPIC_NB_MESSAGES"
+            value = "1000"
+          }
+          # Java specifics
+#          env {
+#            name  = "KAFKA_SASL_JAAS_CONFIG"
+#            value = "org.apache.kafka.common.security.plain.PlainLoginModule   required username='${confluent_api_key.perf-test-client-api-key.id}'   password='${confluent_api_key.perf-test-client-api-key.secret}';"
+#          }
+#          librdkafka specifics
+          env {
+            name = "KAFKA_SASL_MECHANISMS"
+            value = "PLAIN"
+          }
+          env {
+            name = "KAFKA_SASL_USERNAME"
+            value = confluent_api_key.perf-test-client-api-key.id
+          }
+          env {
+            name = "KAFKA_SASL_PASSWORD"
+            value = confluent_api_key.perf-test-client-api-key.secret
+          }
+          env {
+            name = "KAFKA_BATCH_SIZE"
+            value = "100000"
+          }
+          env {
+            name = "KAFKA_LINGER_MS"
+            value = "10"
+          }
+
+          env {
+            name = "KAFKA_BATCH_SIZE"
+            value = "100000"
+          }
+          env {
+            name = "KAFKA_LINGER_MS"
+            value = "10"
+          }
+          env {
+            name = "KAFKA_MAX_IN_FLIGHT_REQUESTS_PER_CONNECTION"
+            value = "5"
+          }
+          env {
+            name = "NB_TOPICS"
+            value = var.nb_topics
+          }
+          env {
+            name = "NUMBER_OF_PARTITIONS"
+            value = var.nb_partitions
+          }
+          env {
+            name = "MESSAGE_SIZE"
+            value = "200"
+          }
+          env {
+            name = "NB_MESSAGES"
+            value = "1000000"
+          }
+          env {
+            name = "USE_RANDOM_KEYS"
+            value = "true"
+          }
+#          env {
+#            name = "RUST_LOG"
+#            value = "DEBUG,librdkafka=trace,rdkafka::client=debug"
+#          }
+        }
+        restart_policy = "Never"
+      }
+    }
+    backoff_limit = 4
+    completions = var.producer_instances
+  }
+  wait_for_completion = true
+  timeouts {
+    create = "5m"
+    update = "5m"
+  }
```

Comment:
This seems to low, I run tests with the java producer and get the following errors : 

> │ Error: job: kafka-producer-benchmark/producer-benchmark is not in complete state
│ 
│   with kubernetes_job.producer-benchmark,
│   on kubernetes.tf line 7, in resource "kubernetes_job" "producer-benchmark":
│    7: resource "kubernetes_job" "producer-benchmark" {

I think this is because the bench takes more than 5min with those parameters.
While looking closely at producer logs (java client) I saw the following : 

> 07:44:15.697 [kafka-producer-network-thread | producer-1] ERROR com.sample.Injector - failed sending in sample_0 key: a57b4e28-8c3d-403a-888e-82b007bd2608
org.apache.kafka.common.errors.TimeoutException: Expiring 403 record(s) for sample_0-5:120000 ms has passed since batch creation
07:44:15.697 [kafka-producer-network-thread | producer-1] ERROR com.sample.Injector - failed sending in sample_0 key: a57b4e28-8c3d-403a-888e-82b007bd2608
org.apache.kafka.common.errors.TimeoutException: Expiring 403 record(s) for sample_0-5:120000 ms has passed since batch creation

Reply:
Sounds fair.